packages/opentelemetry-resources/test/detectors/AwsEc2Detector.test.ts

```diff
@@ -68,18 +80,72 @@ describe('awsEc2Detector', () => {
       assertHostResource(resource, {
         id: 'my-instance-id',
         hostType: 'my-instance-type',
+        name: 'my-hostname',
+        hostName: 'my-hostname',
       });
     });
   });
 
-  describe('with failing request', () => {
-    it('should return empty resource', async () => {
-      const scope = nock(AWS_HOST).get(AWS_PATH).replyWithError({
-        code: 'ENOTFOUND',
+  describe('with unsuccessful request', () => {
+    it('should return empty resource when receiving error response code', async () => {
+      const scope = nock(AWS_HOST)
+        .persist()
+        .put(AWS_TOKEN_PATH)
+        .matchHeader('X-aws-ec2-metadata-token-ttl-seconds', '60')
+        .reply(200, () => mockedTokenResponse)
+        .get(AWS_IDENTITY_PATH)
+        .matchHeader('X-aws-ec2-metadata-token', mockedTokenResponse)
+        .reply(200, () => mockedIdentityResponse)
+        .get(AWS_HOST_PATH)
+        .matchHeader('X-aws-ec2-metadata-token', mockedTokenResponse)
+        .reply(404, () => new Error('NOT FOUND'));
+
+      const resource: Resource = await awsEc2Detector.detect({
+        logger: new NoopLogger(),
+      });
+
+      scope.done();
+
+      assert.ok(resource);
+      assertEmptyResource(resource);
+    });
+
+    it('should return empty resource when timeout', async () => {
+      const scope = nock(AWS_HOST)
+        .put(AWS_TOKEN_PATH)
+        .matchHeader('X-aws-ec2-metadata-token-ttl-seconds', '60')
+        .reply(200, () => mockedTokenResponse)
+        .get(AWS_IDENTITY_PATH)
+        .matchHeader('X-aws-ec2-metadata-token', mockedTokenResponse)
+        .reply(200, () => mockedIdentityResponse)
+        .get(AWS_HOST_PATH)
+        .matchHeader('X-aws-ec2-metadata-token', mockedTokenResponse)
+        .delayConnection(2000)
+        .reply(200, () => mockedHostResponse);
+
+      const resource: Resource = await awsEc2Detector.detect({
+        logger: new NoopLogger(),
       });
+
+      scope.done();
+
+      assert.ok(resource);
+      assertEmptyResource(resource);
+    });
+
+    it('should return empty resource when replied Error', async () => {
+      const scope = nock(AWS_HOST)
+        .put(AWS_TOKEN_PATH)
+        .matchHeader('X-aws-ec2-metadata-token-ttl-seconds', '60')
+        .reply(200, () => mockedTokenResponse)
+        .get(AWS_IDENTITY_PATH)
+        .matchHeader('X-aws-ec2-metadata-token', mockedTokenResponse)
```

Comment:
please reuse the defined const

Reply:
Thank you for your advice, will do it right now!